ReactWindows/ReactNative.Shared.Tests/UIManager/ViewManagerExtensionTests.cs

```diff
@@ -0,0 +1,189 @@
+using Newtonsoft.Json.Linq;
+using NUnit.Framework;
+using ReactNative.UIManager;
+using ReactNative.UIManager.Annotations;
+using System;
+using System.Collections.Generic;
+#if WINDOWS_UWP
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+#else
+using System.Windows;
+using System.Windows.Controls;
+#endif
+
+namespace ReactNative.Tests.UIManager
+{
+    [TestFixture]
+    class ViewManagerExtensionTests
+    {
+        [Test]
+        public void ViewManagerExtension_ArgumentChecks()
+        {
+            var vmx = new TestViewManagerExtension();
+            AssertEx.Throws<ArgumentNullException>(
+                () => vmx.UpdateProperties(null, null),
+                ex => Assert.AreEqual("props", ex.ParamName));
+        }
+
+        [Test]
+        public void ViewManagerExtension_AddEventEmitters()
+        {
+            var vmx = new TestViewManagerExtension();
+            var count = 0;
+            vmx.AddEventEmittersAction = () => count++;
+            vmx.OnCreateView(null, null);
+            Assert.AreEqual(1, count);
+        }
+
+        [Test]
+        public void ViewManagerExtension_NativeProperties()
+        {
+            var vmx = new TestViewManagerExtension();
+            var props = vmx.NativeProperties;
+            Assert.IsTrue(props.ContainsKey("foo"));
```

Comment:
Using Assert.That(props, Does.ContainKey("foo") here will give better failure output when the test fails by showing what keys were actually present and prevent the need to breakpoint or console logging. You could create the expected dictionary ahead and use Is.EquivalentTo(expected), which will also give good error output without the need for discrete assertions.